### ISSUE TITLE
For document clustering, should we leave instruction blank?

### ISSUE THREAD

**Issue by griff4692** [ContextualAI/gritlm#6]
Thanks -- I am using Grit for document embeddings that will be used to score doc-to-doc similarity.

Should I add an instruction or leave it blank?

Thank you,
Griffin

**Comment by Muennighoff** [comment]
> Thanks -- I am using Grit for document embeddings that will be used to score doc-to-doc similarity.
> 
> Should I add an instruction or leave it blank?
> 
> Thank you, Griffin

So is it like STS rather than Retrieval? I would probably add them in that case, but it may make sense to try both.

**Comment by griff4692** [comment]
> > Thanks -- I am using Grit for document embeddings that will be used to score doc-to-doc similarity.
> > Should I add an instruction or leave it blank?
> > Thank you, Griffin
> 
> So is it like STS rather than Retrieval? I would probably add them in that case, but it may make sense to try both.

Thanks for the reply!  Yes - in order to cluster documents for in-context pre-training (https://arxiv.org/abs/2310.10638).

Was going to try "Identify the main topics from a medical document." but wasn't sure how instructions for embeddings are meant to be worded for gritlm.

**Comment by Muennighoff** [comment]
> > > Thanks -- I am using Grit for document embeddings that will be used to score doc-to-doc similarity.
> > > Should I add an instruction or leave it blank?
> > > Thank you, Griffin
> > 
> > 
> > So is it like STS rather than Retrieval? I would probably add them in that case, but it may make sense to try both.
> 
> Thanks for the reply! Yes - in order to cluster documents for in-context pre-training (https://arxiv.org/abs/2310.10638).
> 
> Was going to try "Identify the main topics from a medical document." but wasn't sure how instructions for embeddings are meant to be worded for gritlm.

Yeah I think for clustering you'll get slightly better performance if you include an instruction. The one you proposed sounds good to me!
